From babd10015ebb3ef9e36f5f745d1b5033d5ecb26d Mon Sep 17 00:00:00 2001
From: Ricardo Neri <ricardo.neri@ti.com>
Date: Tue, 8 Mar 2011 15:39:01 -0600
Subject: OMAP4: HDMI: Hack: do not check ACK payload in ACR workaround

There is a known issue in the Ducati HDMI module that implements
the ACR workaround. When the Notify event payload is 0 the payload
of the ACK is not 0 as it should be. Then, verification of the
payload is disabled.

Change-Id: I6fb20ff01ebe5bc1cf7576f7eeb83c98934a8093
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
---
 arch/arm/plat-omap/hdmi_lib.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-omap/hdmi_lib.c b/arch/arm/plat-omap/hdmi_lib.c
index 44b9aa5..14aca6c 100644
--- a/arch/arm/plat-omap/hdmi_lib.c
+++ b/arch/arm/plat-omap/hdmi_lib.c
@@ -1345,8 +1345,19 @@ int hdmi_lib_acr_wa_send_event(u32 payload)
 			tout = schedule_timeout(msecs_to_jiffies(5000));
 			if (!tout)
 				return -EIO;
+
+			/*
+			 * Enable this code when the following patch
+			 * from Ducati MM is released:
+			 * hdmiwa: Reseting properly the hdmi status variable
+			 * It ensures that hdmi_status variable is reset for
+			 * each time a notification is received.
+			 */
+
+#if 0
 			if (payload != hdmi.ack_payload)
 				return -EBADE;
+#endif
 			return 0;
 		}
 		return -ENODEV;
-- 
1.6.0.4

